core-bundle/src/Doctrine/Backup/Backup.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Doctrine\Backup;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Webmozart\PathUtil\Path;
+
+class Backup
+{
+    public const DATETIME_FORMAT = 'YmdHis';
+    public const VALID_BACKUP_NAME_REGEX = '@^.*__(\d{4}\d{2}\d{2}\d{2}\d{2}\d{2})\.sql(\.gz)?$@';
+
+    private string $filepath;
+    private \DateTimeInterface $createdAt;
+
+    public function __construct(string $filepath)
+    {
+        $this->filepath = self::validateFilePath($filepath);
+        $this->createdAt = self::extractDatetime($filepath);
+    }
+
+    /**
+     * Returns the size in bytes.
+     */
+    public function getSize(): int
+    {
+        return (int) filesize($this->getFilepath());
+    }
+
+    public function getHumanReadableSize(): string
+    {
+        $base = log($this->getSize()) / log(1024);
+        $suffix = ['B', 'KB', 'MB', 'GB', 'TB'][floor($base)];
```

Comment:
```suggestion
        $suffix = ['B', 'KiB', 'MiB', 'GiB', 'TiB'][floor($base)];
```

In Contao we use kibibyte etc. 🙃

Reply:
Yeah I actually wanted to reuse `System::getReadableSize()` here but that's still a static method, uses superglobals  etc. We might do that once that was migrated to a proper service as well (cannot be static due to the dep on the translator service).